### ISSUE TITLE
Seems that Sunshine is broken now

### ISSUE THREAD

**Issue by ArtemChandragupta** [NixOS/nixpkgs#343479]
### Describe the bug
I had a Sunshine server set up, but after flake update and rebuild there is no image. Mouse movements and sound still there.

Log of `sunshine` command in terminal (while the server is working, so there may be some bloat)
<details>

```console

[2024:09:21:14:16:34]: Info: Sunshine version: 0.23.1
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: /dev/dri/card1 -> nvidia-drm
[2024:09:21:14:16:34]: Info: System tray created
[2024:09:21:14:16:34]: Error: Environment variable WAYLAND_DISPLAY has not been defined
[2024:09:21:14:16:34]: Info: Detecting monitors
[2024:09:21:14:16:34]: Info: Detected monitor 0: HDMI-0, connected: true
[2024:09:21:14:16:34]: Info: Detected monitor 1: DP-0, connected: false
[2024:09:21:14:16:34]: Info: Detected monitor 2: DP-1, connected: false
[2024:09:21:14:16:34]: Info: Detected monitor 3: DP-2, connected: false
[2024:09:21:14:16:34]: Info: Detected monitor 4: DP-3, connected: false
[2024:09:21:14:16:34]: Info: Detected monitor 5: DP-4, connected: false
[2024:09:21:14:16:34]: Info: Detected monitor 6: DP-5, connected: false
[2024:09:21:14:16:34]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024:09:21:14:16:34]: Info: Trying encoder [nvenc]
[2024:09:21:14:16:34]: Info: Screencasting with KMS
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: /dev/dri/card1 -> nvidia-drm
[2024:09:21:14:16:34]: Error: Couldn't find monitor [0]
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: /dev/dri/card1 -> nvidia-drm
[2024:09:21:14:16:34]: Error: Couldn't find monitor [0]
[2024:09:21:14:16:34]: Info: Screencasting with KMS
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: /dev/dri/card1 -> nvidia-drm
[2024:09:21:14:16:34]: Error: Couldn't find monitor [0]
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: /dev/dri/card1 -> nvidia-drm
[2024:09:21:14:16:34]: Error: Couldn't find monitor [0]
[2024:09:21:14:16:34]: Info: Encoder [nvenc] failed
[2024:09:21:14:16:34]: Info: Trying encoder [vaapi]
[2024:09:21:14:16:34]: Info: Screencasting with KMS
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
libva info: VA-API version 1.22.0
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: Found monitor for DRM screencasting
[2024:09:21:14:16:34]: Error: Failed to determine panel orientation, defaulting to landscape.
[2024:09:21:14:16:34]: Info: Found connector ID [36]
[2024:09:21:14:16:34]: Warning: No KMS cursor plane found. Cursor may not be displayed while streaming!
libva info: Trying to open /run/opengl-driver/lib/dri/simpledrm_drv_video.so
libva info: Trying to open /usr/lib/dri/simpledrm_drv_video.so
libva info: Trying to open /usr/lib32/dri/simpledrm_drv_video.so
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/simpledrm_drv_video.so
libva info: Trying to open /usr/lib/i386-linux-gnu/dri/simpledrm_drv_video.so
libva info: va_openDriver() returns -1
[2024:09:21:14:16:34]: Error: Couldn't initialize va display: unknown libva error
[2024:09:21:14:16:34]: Warning: Monitor  doesn't support hardware encoding. Reverting back to GPU -> RAM -> GPU
[2024:09:21:14:16:34]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:34]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:34]: Info: Found monitor for DRM screencasting
[2024:09:21:14:16:34]: Error: Failed to determine panel orientation, defaulting to landscape.
[2024:09:21:14:16:34]: Info: Found connector ID [36]
[2024:09:21:14:16:34]: Warning: No KMS cursor plane found. Cursor may not be displayed while streaming!
[2024:09:21:14:16:35]: Info: SDR color coding [Rec. 601]
[2024:09:21:14:16:35]: Info: Color depth: 8-bit
[2024:09:21:14:16:35]: Info: Color range: [JPEG]
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0
[2024:09:21:14:16:35]: Error: [h264_vaapi @ 0x2bba05c0] No usable encoding entrypoint found for profile VAProfileH264High (7).
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
[2024:09:21:14:16:35]: Info: Retrying with fallback configuration options for [h264_vaapi] after error: Function not implemented
libva info: va_openDriver() returns 0
[2024:09:21:14:16:35]: Error: [h264_vaapi @ 0x2bbdf680] No usable encoding entrypoint found for profile VAProfileH264High (7).
[2024:09:21:14:16:35]: Error: Could not open codec [h264_vaapi]: Function not implemented
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
[2024:09:21:14:16:35]: Info: SDR color coding [Rec. 601]
[2024:09:21:14:16:35]: Info: Color depth: 8-bit
[2024:09:21:14:16:35]: Info: Color range: [JPEG]
libva info: va_openDriver() returns 0
[2024:09:21:14:16:35]: Error: [h264_vaapi @ 0x2bbdf680] No usable encoding entrypoint found for profile VAProfileH264High (7).
[2024:09:21:14:16:35]: Info: Retrying with fallback configuration options for [h264_vaapi] after error: Function not implemented
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0
[2024:09:21:14:16:35]: Error: [h264_vaapi @ 0x2bba05c0] No usable encoding entrypoint found for profile VAProfileH264High (7).
[2024:09:21:14:16:35]: Error: Could not open codec [h264_vaapi]: Function not implemented
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
[2024:09:21:14:16:35]: Info: SDR color coding [Rec. 601]
[2024:09:21:14:16:35]: Info: Color depth: 8-bit
[2024:09:21:14:16:35]: Info: Color range: [JPEG]
libva info: va_openDriver() returns 0
[2024:09:21:14:16:36]: Error: [h264_vaapi @ 0x2bba05c0] No usable encoding entrypoint found for profile VAProfileH264High (7).
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
[2024:09:21:14:16:36]: Info: Retrying with fallback configuration options for [h264_vaapi] after error: Function not implemented
libva info: va_openDriver() returns 0
[2024:09:21:14:16:36]: Error: [h264_vaapi @ 0x2bbdf680] No usable encoding entrypoint found for profile VAProfileH264High (7).
[2024:09:21:14:16:36]: Error: Could not open codec [h264_vaapi]: Function not implemented
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
[2024:09:21:14:16:36]: Info: SDR color coding [Rec. 601]
[2024:09:21:14:16:36]: Info: Color depth: 8-bit
[2024:09:21:14:16:36]: Info: Color range: [JPEG]
libva info: va_openDriver() returns 0
[2024:09:21:14:16:36]: Error: [h264_vaapi @ 0x2bbdf680] No usable encoding entrypoint found for profile VAProfileH264High (7).
[2024:09:21:14:16:36]: Info: Retrying with fallback configuration options for [h264_vaapi] after error: Function not implemented
libva info: VA-API version 1.22.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0
[2024:09:21:14:16:36]: Error: [h264_vaapi @ 0x2bba05c0] No usable encoding entrypoint found for profile VAProfileH264High (7).
[2024:09:21:14:16:36]: Error: Could not open codec [h264_vaapi]: Function not implemented
[2024:09:21:14:16:36]: Info: Encoder [vaapi] failed
[2024:09:21:14:16:36]: Info: Trying encoder [software]
[2024:09:21:14:16:36]: Info: Screencasting with KMS
[2024:09:21:14:16:36]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:36]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:36]: Info: Found monitor for DRM screencasting
[2024:09:21:14:16:36]: Error: Failed to determine panel orientation, defaulting to landscape.
[2024:09:21:14:16:36]: Info: Found connector ID [36]
[2024:09:21:14:16:36]: Warning: No KMS cursor plane found. Cursor may not be displayed while streaming!
[2024:09:21:14:16:36]: Info: SDR color coding [Rec. 601]
[2024:09:21:14:16:36]: Info: Color depth: 8-bit
[2024:09:21:14:16:36]: Info: Color range: [JPEG]
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] profile High, level 4.2, 4:2:0, 8-bit
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] frame I:1     Avg QP:31,00  size:  1203
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] mb I  I16..4: 99,9%  0,0%  0,0%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] 8x8 transform intra:0,0%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] coded y,uvDC,uvAC intra: 0,0% 0,0% 0,0%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] i16 v,h,dc,p: 97%  0%  3%  0%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu:  0%  0% 75% 12%  0%  0%  0%  0% 12%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu:  0%  0% 100%  0%  0%  0%  0%  0%  0%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] i8c dc,h,v,p: 100%  0%  0%  0%
[2024:09:21:14:16:36]: Info: [libx264 @ 0x2bba05c0] kb/s:577,44
[2024:09:21:14:16:36]: Info: Screencasting with KMS
[2024:09:21:14:16:36]: Info: /dev/dri/card0 -> simpledrm
[2024:09:21:14:16:36]: Warning: No render device name for: /dev/dri/card0
[2024:09:21:14:16:36]: Info: Found monitor for DRM screencasting
[2024:09:21:14:16:36]: Error: Failed to determine panel orientation, defaulting to landscape.
[2024:09:21:14:16:36]: Info: Found connector ID [36]
[2024:09:21:14:16:36]: Warning: No KMS cursor plane found. Cursor may not be displayed while streaming!
[2024:09:21:14:16:36]: Info: 
[2024:09:21:14:16:36]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024:09:21:14:16:36]: Info: 
[2024:09:21:14:16:36]: Info: Found H.264 encoder: libx264 [software]
[2024:09:21:14:16:36]: Fatal: Couldn't bind RTSP server to port [48010], Address already in use
[2024:09:21:14:16:36]: Info: Adding avahi service Sunshine

```

</details

To ensure that it is not because of sunshine state, I commented out sunshine module, rebuilded, collected garbage, removed sunshine config from `~/config` and included it again. After new configuration in web interface, it behaves the same (no image).

I use Gnome-xorg on Nvidia and nixos-unstable with flakes. [This is my config](https://github.com/ArtemChandragupta/NixFlake)

### Steps To Reproduce
1. Have Nvidia and Gnome xorg
2. do
```nix
{

services.sunshine = {
  enable = true;
  autoStart = true;
  capSysAdmin = true;
  openFirewall = true;
};

}
```

### Expected behavior
I should get an image output also of course

### Notify maintainers
@devusb

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.0, NixOS, 24.11 (Vicuna), 24.11.20240919.c04d565`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/hiasfhl8f5yy88hcfbr3s8s4bm63wsjw-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by ArtemChandragupta** [comment]
Update:

On Gnome Wayland it is working.

**Comment by devusb** [comment]
Haven't used NVIDIA + xorg in awhile -- try without `capSysAdmin = true`?

**Comment by ArtemChandragupta** [comment]
Can you explain what is "awhile -- try" and "capSysAdmin = true"? I never encountered this and simple googling gives bad results.

**Comment by devusb** [comment]
In your above configuration you show that you used 
```
services.sunshine = {
  enable = true;
  autoStart = true;
  capSysAdmin = true;
  openFirewall = true;
};
```

remove the`capSysAdmin` line such that it becomes

```
services.sunshine = {
  enable = true;
  autoStart = true;
  openFirewall = true;
};
```
and see if you have different behavior.

I don't have an NVIDIA card anymore so it's difficult for me to test this setup -- I did use Plasma + xorg + NVIDIA for awhile and it generally worked fine, but thinking that I only needed `capSysAdmin` for KMS when using Wayland.

**Comment by ArtemChandragupta** [comment]
Yes, this fixes the issue.

So, I think it is needed to make a comment about this in both NixOS wiki.

**Comment by devusb** [comment]
@ArtemChandragupta good point -- I updated the wiki page. thanks for the feedback.

**Comment by ArtemChandragupta** [comment]
Hi. It's sad to say, but I misled you. Unfortunately, this "solution" resulted in the stream starting time to time, depending on the session. Same with wayland gnome and hyprland, time to time.

I carried out testing, so to speak, and found out that the problem is in the newest kernel/Nvidia drivers, which are updated along with it, namely

`kernelPackages = pkgs.linuxPackages_latest;`

Removing it I got reliable streaming. I waited threw one default kernel update and three latest kernel updates to get sure that I won't make mistake in this time.
